build/ci/job-template.yml

```diff
@@ -74,8 +75,35 @@ jobs:
           - script: ${{ parameters.buildScript }} -$(_configuration) -runtests -coverage=${{ parameters.codeCoverage }}
             displayName: Run All Tests.
         - ${{ if eq(parameters.runSpecific, 'true') }}:
-          - script: ${{ parameters.buildScript }} -$(_configuration) -runSpecificTests -coverage=${{ parameters.codeCoverage }}
-            displayName: Run Specific Tests.
+          - ${{ if and(eq(parameters.useVSTestTask, 'true'), eq(parameters.buildScript, 'build.cmd')) }}:
+            - bash: echo "##vso[task.setvariable variable=PROCDUMP_PATH]D:/a/1/s/tools-local/procdump"
```

Comment:
>D:/a/1/s/tools-local/procdump [](start = 72, length = 29)

this path is not reliable, like in test agent the path is different, could you investigate why $(Build.SourcesDirectory) not working?

Reply:
"Not reliable" is to put it mildly. Please do not hard code paths ever.

Reply:
I was having issues with defining PROCDUMP_PATH in job-template.yml, hence the hardcode here. I fixed it by defining PROCDUMP_PATH in the `variables `list in job-template.yml. Apparently one can declare environment variables directly in .yml, and there is no need for a `echo "##vso[task.setvariable variable=PROCDUMP_PATH]D:/a/1/s/tools-local/procdump"` .